code/engine/xrCore/xrMemory.cpp

```diff
@@ -119,412 +36,25 @@ void xrMemory::mem_compact() {
-ICF u32 get_pool(size_t size) {
-    u32 pid = u32(size / mem_pools_ebase);
-    if (pid >= mem_pools_count)
-        return mem_generic;
-    else
-        return pid;
-}
-
-#ifdef PURE_ALLOC
-bool g_use_pure_alloc = false;
-#endif // PURE_ALLOC
-
 void* xrMemory::mem_alloc(size_t size) {
-    stat_calls++;
-
-#ifdef PURE_ALLOC
-    static bool g_use_pure_alloc_initialized = false;
-    if (!g_use_pure_alloc_initialized) {
-        g_use_pure_alloc_initialized = true;
-        g_use_pure_alloc =
-#ifdef XRCORE_STATIC
-            true
-#else  // XRCORE_STATIC
-            !!strstr(GetCommandLine(), "-pure_alloc")
-#endif // XRCORE_STATIC
-            ;
-    }
-
-    if (g_use_pure_alloc) {
-        void* result = malloc(size);
-#ifdef USE_MEMORY_MONITOR
-        memory_monitor::monitor_alloc(result, size, _name);
-#endif // USE_MEMORY_MONITOR
-        return (result);
-    }
-#endif // PURE_ALLOC
-
-#ifdef DEBUG_MEMORY_MANAGER
-    if (mem_initialized)
-        debug_cs.Enter();
-#endif // DEBUG_MEMORY_MANAGER
-
-    u32 _footer = debug_mode ? 4 : 0;
-    void* _ptr = 0;
-
-    //
-    if (!mem_initialized /*|| debug_mode*/) {
-        // generic
-        //	Igor: Reserve 1 byte for xrMemory header
-        void* _real = xr_aligned_offset_malloc(1 + size + _footer, 16, 0x1);
-        // void*	_real			=	xr_aligned_offset_malloc	(size + _footer,
-        // 16, 0x1);
-        _ptr = (void*)(((u8*)_real) + 1);
-        *acc_header(_ptr) = mem_generic;
-    } else {
-#ifdef DEBUG_MEMORY_MANAGER
-        save_stack_trace();
-#endif // DEBUG
-       //	accelerated
-       //	Igor: Reserve 1 byte for xrMemory header
-        u32 pool = get_pool(1 + size + _footer);
-        // u32	pool				=	get_pool	(size+_footer);
-        if (mem_generic == pool) {
-            // generic
-            //	Igor: Reserve 1 byte for xrMemory header
-            void* _real = xr_aligned_offset_malloc(1 + size + _footer, 16, 0x1);
-            // void*	_real		=	xr_aligned_offset_malloc	(size +
-            // _footer,16,0x1);
-            _ptr = (void*)(((u8*)_real) + 1);
-            *acc_header(_ptr) = mem_generic;
-        } else {
-            // pooled
-            //	Igor: Reserve 1 byte for xrMemory header
-            //	Already reserved when getting pool id
-            void* _real = mem_pools[pool].create();
-            _ptr = (void*)(((u8*)_real) + 1);
-            *acc_header(_ptr) = (u8)pool;
-        }
-    }
-
-#ifdef DEBUG_MEMORY_MANAGER
-    if (debug_mode)
-        dbg_register(_ptr, size, _name);
-    if (mem_initialized)
-        debug_cs.Leave();
-        // if(g_globalCheckAddr==_ptr){
-        //	__debugbreak();
-        //}
-        // if (_name && (0==strcmp(_name,"class ISpatial *")) && (size==376))
-        //{
-        //	__debugbreak();
-        //}
-#endif // DEBUG_MEMORY_MANAGER
-#ifdef USE_MEMORY_MONITOR
-    memory_monitor::monitor_alloc(_ptr, size, _name);
-#endif // USE_MEMORY_MONITOR
-    return _ptr;
+    void* ptr = malloc(size);
```

Comment:
simplify: return malloc(size);

Reply:
it is for later to implement tbb; compiler for now will simplify

Reply:
anyway this code should be completely removed to use tbb allocator

Reply:
actually I want to use this code for tbb allocator. xl_memory functions will call tbb functions

Reply:
I don't see any reason to keep that boolshit. tbb allocator could be included as a proxy library without any additional work

Reply:
For now, this is not the point of PR